pkgs/by-name/tr/treefmt/build-config.nix

```diff
@@ -0,0 +1,26 @@
+{
+  lib,
+  formats,
+}:
+module:
+let
+  settingsFormat = formats.toml { };
+  configuration = lib.evalModules {
+    modules = [
+      {
+        freeformType = settingsFormat.type;
+      }
+      {
+        _file = "<treefmt2.buildConfig args>";
```

Comment:
Best to use ./build-config.nix to provide the actual context.

Reply:
Normally I'd agree, but in this case we don't want `./build-config.nix` to be the `_file`.

Instead, we're wrapping the user-supplied modules and setting a (default) file location that indicates the module was supplied to this function (`treefmt.buildConfig`).

I've added some comments for clarity and fixed a typo (`treefmt2` -> `treefmt`).